diff --git a/Build.PL b/Build.PL
index 3c03f92..2258293 100644
--- a/Build.PL
+++ b/Build.PL
@@ -26,18 +26,23 @@ sub ACTION_install {
     my $self = shift;
     my $dir  = $self->base_dir();
     $self->SUPER::ACTION_install();
-    mkdir '/var/lib/lsm';
+    
+    my $destdir = $self->destdir || $ENV{DESTDIR} || '';
+    
+    mkdir "$destdir/var/lib/lsm";
     my $path = $self->install_destination('script');
     my $install_etc = Net::ISP::Balance->install_etc;
-    unlink("$install_etc/load_balance.pl");
-    symlink("$path/load_balance.pl"=>"$install_etc/load_balance.pl");
-    # fix permissions of balance.conf, to encourage root to edit
-    chmod 0644,"$install_etc/balance.conf";
-    # and rules files
-    system "chmod u+w $install_etc/balance/firewall/*";
-    system "chmod u+w $install_etc/balance/routes/*";
-    system "chmod u+w $install_etc/balance/pre-run/*";
-    system "chmod u+w $install_etc/balance/post-run/*";
+    
+    system "mkdir -p $destdir$install_etc";
+    system "cp etc/balance.conf $destdir$install_etc/";
+    system "cp -a etc/balance $destdir$install_etc/";
+    
+    chmod 0755, glob("$destdir$install_etc/balance/firewall/*");
+    chmod 0755, glob("$destdir$install_etc/balance/routes/*");
+    chmod 0755, glob("$destdir$install_etc/balance/pre-run/*");
+    chmod 0755, glob("$destdir$install_etc/balance/post-run/*");
+
+    system "ln -sf $path/load_balance.pl $destdir/etc/load_balance.pl" unless $destdir;
 }
 
 sub process_lsm_files {
diff --git a/lib/Net/ISP/Balance.pm b/lib/Net/ISP/Balance.pm
index 10e5f5b..fd5af0f 100644
--- a/lib/Net/ISP/Balance.pm
+++ b/lib/Net/ISP/Balance.pm
@@ -1237,6 +1237,7 @@ RedHat/CentOS network configuration files in
 
 sub install_etc {
     my $self = shift;
+    return '/etc/net-isp-balance'            if -e '/etc/gentoo-release' || -d '/etc/net-isp-balance';
     return '/etc/network'                    if -d '/etc/network';
     return '/etc/sysconfig/network-scripts'  if -d '/etc/sysconfig/network-scripts';
     return '/etc';
diff --git a/lsm/Makefile b/lsm/Makefile
index 99d7d66..5f25775 100644
--- a/lsm/Makefile
+++ b/lsm/Makefile
@@ -8,8 +8,8 @@ VERSION	?= $(lastword $(shell grep ^Version: foolsm.spec))
 PROGS	= foolsm
 PKG     = foolsm
 
-CC	= gcc
-override CFLAGS	+= -w -O2 -DFOOLSM_VERSION=\"$(VERSION)\"
+CC	?= gcc
+CFLAGS	+= -w -O2 -DFOOLSM_VERSION=\"$(VERSION)\"
 #override CFLAGS += -D NO_PLUGIN_EXPORT
 #override CFLAGS += -D NO_PLUGIN_EXPORT_MUNIN
 #override CFLAGS += -D NO_PLUGIN_EXPORT_STATUS
